### DISCUSSION TITLE
Help: Upgrading some APIs of webpack did not find a replacement

### DISCUSSION THREAD

**Discussion by hcxmj** [webpack/webpack#15983] [General]
"webpack": "^5.73.0",
    "webpack-cli": "^4.9.2",
    "webpack-dev-server": "^4.9.2",

When I upgrade to the above version，prompt me “devServer.use is not a function”.
Below is my code writing:

![image](https://user-images.githubusercontent.com/58494807/175845146-dd94309c-8b18-41a2-91db-c86f979f3dbd.png)
![image](https://user-images.githubusercontent.com/58494807/175845208-72cc9c79-86f0-490c-bd62-dc336ec2118c.png)

**Comment by vankop** [comment]
@alexander-akait any ideas?

**Reply by alexander-akait** [reply]
We just move internal server under `app`, i.e. `devServer.app.use` should work.

Anyway it is wrong approach, we have https://webpack.js.org/configuration/dev-server/#devserversetupmiddlewares for this

**Reply by hcxmj** [reply]
ok, thank you @vankop and @alexander-akait ,I try to use setupMiddlewares first,If it doesn't work, try to rewrite the configuration file, because the version span is a bit large
